packages/e2e-tests/src/tests/SessionSigs.spec.ts

```diff
@@ -0,0 +1,825 @@
+    });
+
+    it('LitAction IPFS Session', async () => {
+      await decryptAndCombine(devEnv, getLitActionSessionSigsUsingIpfsId);
+    });
+
+    it('EOA Wallet', async () => {
+      await decryptAndCombine(devEnv, getEoaSessionSigs);
+    });
+
+    it('PKP Session', async () => {
+      await decryptAndCombine(devEnv, getPkpSessionSigs);
+    });
+  });
+
+  describe('Sign And Combine ECDSA', () => {
+    it('LitAction Session', async () => {
+      await signAndCombine(devEnv, getLitActionSessionSigs);
+    });
+
+    it('LitAction IPFS Session', async () => {
+      await signAndCombine(devEnv, getLitActionSessionSigsUsingIpfsId);
+    });
+
+    it('EOA Wallet', async () => {
+      await signAndCombine(devEnv, getEoaSessionSigs);
+    });
+
+    it('PKP Session', async () => {
+      await signAndCombine(devEnv, getPkpSessionSigs);
+    });
+  });
+
+  it('Invalid lit action Custom Auth SessionSigs', async () => {
+    devEnv.setUnavailable(LIT_TESTNET.MANZANO);
+
+    const alice = await devEnv.createRandomPerson();
+
+    try {
+      await getInvalidLitActionSessionSigs(devEnv, alice);
+    } catch (e: any) {
+      console.log('❌ This error is expected', e);
+      if (
+        e.message ===
+        'There was an error getting the signing shares from the nodes'
+      ) {
+        console.log(
+          '✅ testUseInvalidLitActionCodeToGenerateSessionSigs passed'
+        );
+      } else {
+        throw e;
+      }
+    } finally {
+      devEnv.releasePrivateKeyFromUser(alice);
+    }
+  });
+
+  it('Invalid Lit Action Custom Auth IPFS SessionSigs', async () => {
+    const alice = await devEnv.createRandomPerson();
+
+    try {
+      await getInvalidLitActionIpfsSessionSigs(devEnv, alice);
+    } catch (e: any) {
+      console.log('❌ THIS IS EXPECTED: ', e);
+
+      if (e.message === 'An error related to validation has occured.') {
+        console.log(
+          '✅ testUseInvalidLitActionIpfsCodeToGenerateSessionSigs is expected to have an error'
+        );
+      } else {
+        throw e;
+      }
+    } finally {
+      devEnv.releasePrivateKeyFromUser(alice);
+    }
+  });
+});
+
+const executeJsClaimKeys = async (
+  devEnv: TinnyEnvironment,
+  generator: (
+    devEnv: TinnyEnvironment,
+    person: TinnyPerson,
+    resources?: LitResourceAbilityRequest[]
+  ) => Promise<SessionSigsMap | undefined>
+): Promise<void> => {
+  const alice = await devEnv.createRandomPerson();
+  const litActionSessionSigs = await generator(devEnv, alice, [
+    {
+      resource: new LitPKPResource('*'),
+      ability: LitAbility.PKPSigning,
+    },
+    {
+      resource: new LitActionResource('*'),
+      ability: LitAbility.LitActionExecution,
+    },
+  ]);
+
+  const res = await devEnv.litNodeClient?.executeJs({
+    sessionSigs: litActionSessionSigs,
+    code: `(async () => {
+    Lit.Actions.claimKey({keyId: "foo"});
+    Lit.Actions.claimKey({keyId: "bar"});
+  })();`,
+  });
+
+  devEnv.releasePrivateKeyFromUser(alice);
```

Comment:
Need this to be in a `finally()` block to ensure it is released if any of the code from L288 until this line throws an error